### PR TITLE
Added scale op FP32/BF16 FWD/BWD kernels

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
New features

### PR changes
OPs

### Describe
Added scale op FP32/BF16 FWD/BWD kernels for enabling Word2Vec bf16 training and fixing customer issues with switching from pure PaddlePaddle to oneDNN ops.

### CODE REVIEW

File: paddle/fluid/operators/scale_op.cc [hunk 1]
```diff
@@ -54,6 +54,22 @@ class ScaleOp : public framework::OperatorWithKernel {
...
     ctx->ShareLoD("X", /*->*/ "Out");
   }
+
+  framework::OpKernelType GetExpectedKernelType(
+      const framework::ExecutionContext &ctx) const override {
+    auto input_data_type =
+        framework::OperatorWithKernel::IndicateVarDataType(ctx, "X");
+
+#ifdef PADDLE_WITH_MKLDNN
+    using mkldnn::memory;
```
Comment (lidanqing-vv) [inline paddle/fluid/operators/scale_op.cc]: Maybe mkldnn::memory is not used in this function?

File: paddle/fluid/operators/mkldnn/scale_mkldnn_op.cc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    auto x_tz = framework::vectorize<int64_t>(x->dims());
+    bool is_inplaced = x->IsSharedBufferWith(*out);
+
+    platform::ActivationMKLDNNHandler<T> handler(
+        x_tz, mkldnn::algorithm::eltwise_linear, scale, bias, x->format(),
+        dev_ctx, ctx.GetPlace(), ctx.InputName("X"), is_inplaced);
+
+    auto src_memory_p = handler.AcquireSrcMemory(x);
+    auto dst_memory_p =
+        is_inplaced ? src_memory_p : handler.AcquireDstMemory(out);
```
Comment (jczaja) [inline paddle/fluid/operators/mkldnn/scale_mkldnn_op.cc]: Scale operator of PaddlePaddle does not support inplace execution, so you can drop this inplace part

File: paddle/fluid/operators/scale_op.cc (hunk 1 above)
Comment (jakpiase) [reply paddle/fluid/operators/scale_op.cc]: You're right, thank you I have missed that

Comment (jakpiase) [reply paddle/fluid/operators/scale_op.cc:64]: You're right, I have missed that, thank you

**Review by jczaja — DISMISSED** [review]
LGTM

**Comment by lidanqing-vv** [comment]
@jakpiase Hi did you test with ssdlite_mobilenet_v3 model?

**Review by jczaja — APPROVED** [review]

**Review by lidanqing-vv — APPROVED** [review]
LGTM

**Comment by jakpiase** [comment]
@luotao1 Could you please start your review?

**Comment by lidanqing-vv** [comment]
Hi, 
1. oneDNN scale kernel is found called in DNNL_VERBOSE logs
```
dnnl_verbose,exec,cpu,eltwise,jit:avx512_common,forward_training,data_f32::blocked:acdb:f0 diff_undef::undef::f0,,alg:eltwise_linear alpha:1 beta:3,1x16x160x160,1.36597
```
2. and performance of ssdlite_mobilenetv3 on my i9 machine is improved **from 2.23 QPS to 41.32 QPS.** 
3. No oneDNN ref version is called from the log so perf is good now.

Overall, this PR improve ssdlite_mobilenetv3 performance a lot. We should merge it.

**Review by luotao1 — APPROVED** [review]

**Review by Avin0323 — APPROVED** [review]
LGTM for unity_build_rule.cmake

**Pull request merged**